lib/component-repository/src/routes/virtual-components/UpdateComponentConfig.js

```diff
@@ -1,3 +1,33 @@
+const ComponentConfig = require('../../models/ComponentConfig');
+const mongoose = require('mongoose');
 module.exports = async function (req, res, next) {
-    return next();
+  const { authClientId } = req.body;
+  const { componentVersionId } = req.params;
+  const tenant = req.user.isAdmin ? req.body.tenant : req.user.tenant;
+  if (!mongoose.Types.ObjectId.isValid(componentVersionId)) {
+    return res
+      .status(400)
+      .send({ errors: [{ message: 'Invalid id', code: 400 }] });
+  }
+
+  const qry = { componentVersionId, tenant };
+  const currentVersionComponentConfig = await ComponentConfig.findOne(qry).lean();
```

Comment:
Here, if you don't use lean() and you return the mongoose object, you can then update the properties and call to the currentVersionComponentConfig.save() method to not use findOne again